engine/src/engine.ts

```diff
@@ -1096,13 +1106,17 @@ export default class Engine {
 
   beginSetupBoardPhase() {
     this.changePhase(Phase.SetupBoard);
-    // The last player is the one to rotate the sectors
-    this.currentPlayer = this.players.slice(-1).pop().player;
 
-    if (!this.options.advancedRules) {
-      // No sector rotation
+    if (this.options.customBoardSetup) {
+      initCustomSetup(this);
+
+      // The first player (host) does board setup and rotation
+      this.currentPlayer = this.players[0].player;
+    } else if (this.options.advancedRules) {
+      // The last player is the one to rotate the sectors
+      this.currentPlayer = this.players.slice(-1).pop().player;
+    } else {
```

Comment:
not a change request but instead of changing everything you can do:

```
if (this.options.customboardSetup) {
  ...
  return;
}

// keep same code
```

Reply:
I was a bit confused by the original structure - that's why I changed it.